docs/component.md

```diff
@@ -3,353 +3,581 @@ id: component
-```javascript
-import Component, { html } from '@biotope/element';
-```
-
-## attributes / props
-Every component has its own props. The props are the result of picking and parsing of all the
-attributes of the component.
+## Rendering
+The `render` function was designed to complement the concept of Web Components and take it to the
+next level. It provides you with a simple and intuitive way to render information inside the
+component.
 
-You can pass new props or change the old ones using the attributes you already know from native js
-and html - i.e. by using `setAttribute` and `removeAttribute`. If your component is watching the
-attribute(s) you added/modified, it will parse it/them and trigger a re-render of the component.
+Every component implements a `render` function. Inside it, you need to call and return the `this.html`
+function on a template literal to add HTML to your component:
 
 ```javascript
+// my-button.js
 import Component from '@biotope/element';
 
 class MyButton extends Component {
   render() {
+    // this will add 'Hello World 🐤' to the root of the element
     return this.html`
       <div>
-        🎰 ${this.props.foo}
+        Hello World 🐤
       </div>
     `;
   }
 }
 
 MyButton.componentName = 'my-button';
-MyButton.attributes = ['foo'];
 MyButton.register();
 ```
 
-```html
-<my-button foo="bar"></my-button>
-```
-
-This will result in the following html:
+This new HTML code will be added to the shadow DOM (think of this process as something similar to
+writing a string to the `innerHTML` property of a HTMLElement). The result of the code above is the
+following:
 
 ```html
 <my-button>
-  <div>🎰 bar<div>
+  <div>
+    Hello World 🐤
+  <div>
 </my-button>
 ```
 
-You can also use interpolated attributes, like so:
+> __⚠️ Important:__ You need to create the `render` function when you define your class so that it is
+already available during the component registration phase. This means you **cannot** simply do:
 
 ```javascript
+// my-button.js
+import Component from '@biotope/element';
+import { template } from './template';
+
 class MyButton extends Component {
-  render() {
-    return this.html`
-      <div class="fancy-${someCondition ? 'thing' : 'stuff'}">
-        🎰 ${this.props.foo}
-      </div>
-    `;
-  }
+  // this won't work
+  render = template
 }
+
+MyButton.componentName = 'my-button';
+MyButton.register();
+```
+
+If you implement the example above, the `render` function will only be set during the constructor,
+and not when calling the `register` function. This will also cause your component to render nothing.
+
+### Partials
+The `html` function can be used to convert string HTML to actual HTML components. In this sense, if
+you need to render a non-HTML string, then you can simply interpolate it inside the template string
+that you pass to the `this.html` function. You can even use regular Javascript - inline `if`s are
```

Comment:
> Javascript

Should be:

> JavaScript

Reply:
> inline `if`s

Could be:

> inline conditions

Or something similar, since you're not actually using `if` and it's easier to read.